diaphora_ida.py

```diff
@@ -2806,6 +2845,51 @@ def main():
       os.remove("%s-crash" % file_out)
 
     idaapi.qexit(0)
+  # EXPORT - works with pseudocode, better than above
+  elif os.getenv("DIAPHORA_AUTO2") is not None:
+    if g_debug:
+      log("Handling DIAPHORA_AUTO2")
+      log("DIAPHORA_EXPORT_FILE=%s" % os.getenv("DIAPHORA_EXPORT_FILE"))
+    file_out = os.getenv("DIAPHORA_EXPORT_FILE")
+    if file_out is None:
+      raise Exception("No export file specified!")
+    _diff_or_export(False, file_out=file_out)
+    idaapi.qexit(0)
+  # DIFF-SHOW
+  elif os.getenv("DIAPHORA_AUTO4") is not None:
```

Comment:
To be honest, I don't recall 100% why I used that method (it was a couple of months ago) but it could be because the method in diaphora.py wasn't working for me to do what I wanted? So maybe what I am doing is different from what yo used to do?

So just so I understand, you mean using diaphora.py and DIAPHORA_AUTO_DIFF instead of diaphora_ida.py and DIAPHORA_AUTO2? But if that is the case, diaphora.py calls into diaphora_ida.py anyway so relies on running in IDA, right?

Or you mean using the default case? But then I use it in idahunt for doing the diffing between the 2 databases indeed, since your code did the job for me.

However for DIAPHORA_AUTO2, I am using IDA to export a .sqlite for a given binary, I am not using it for doing the actual diff.

Sorry if I misunderstand what you are asking for...

Reply:
I was referring to the diffing process. So, it's possible to call `diaphora.py` and do the diffing (but not the export) from outside IDA. It doesn't need IDA at all and you can even do it on the command line like this:

```$ /path/to/diaphora.py -o results.db sample1.db sample2.db 
````

Now, considering your changes, I think you decided to add this code because the code responsible of showing the differences in assembler/pseudocode are in `diaphora_ida.py`, right? I believe this is something I must change myself because this functionality is not IDA related and it can be highly useful outside, like in your use case.

Reply:
Regarding the diffing process, I don't use IDA but just use Python afaict, see https://github.com/nccgroup/idahunt#diffing-files

```
[idahunt] C:\Program Files\Python39\python.exe C:\diaphora\diaphora.py C:\tests\patch\tm.sqlite C:\tests\vuln\tm.sqlite -o C:\tests\vuln\patch_vs_vuln\tm.sys.diaphora
```

But I use IDA for the export (before the diff) and also for generating the HTML for assembly and pseudocode (old `DIAPHORA_AUTO4`, now named `DIAPHORA_AUTO_HTML`). Are you saying that we can generate the HTML without using IDA and that that it is already stored in the `results.db`?

Reply:
The HTML is not in the output database, as in the example `results.db`, but the whole source of both functions is, so the diffing can be done outside. But, don't worry, making it more clear is my job.